backend/docs/introducing_new_tools.md

```diff
@@ -0,0 +1,129 @@
+Introducing new tools
+------------------------------------
+
+Here you can find every information that will be helpful while introducing new type of tools to backend. This tutorial assumes that you know how to run MedTagger. If not, go [here](development_setup_native.md) to learn how to run it natively, or [here](development_in_vagrant.md) to learn how to run it in Vagrant.
+
+Adding new tool to MedTagger requires making changes both in backend and in frontend. To see how to add your tool on the frontend side proceed [here](). 
+
+### Step 1. Add tool to the definitions
+
+ Add your new tool to the `medtagger/definitions.py`. The `LabelTool` enum defines all tools supported by MedTagger.
+
+### Step 2. Create tool database model
+
+To enable support for various labels types we have introduced a concept of `LabelElement` which can be seen in `medtagger/database/models.py/`. It represents a high level LabelElement after which more specific types of LabelElement can inherit from. You can read more [here](http://docs.sqlalchemy.org/en/latest/orm/inheritance.html)
```

Comment:
- various **label** types*
  - It represents a high level LabelElement, which more specific types of LabelElement can inherit from.